internal/controller/provisioner/handler.go

```diff
@@ -62,7 +63,7 @@ func (h *eventHandler) HandleEventBatch(ctx context.Context, logger logr.Logger,
 			case *gatewayv1.Gateway:
 				h.store.updateGateway(obj)
 			case *appsv1.Deployment, *appsv1.DaemonSet, *corev1.ServiceAccount,
-				*corev1.ConfigMap, *rbacv1.Role, *rbacv1.RoleBinding:
+				*corev1.ConfigMap, *rbacv1.Role, *rbacv1.RoleBinding, *autoscalingv2.HorizontalPodAutoscaler:
```

Comment:
The reason we include these objects in the Update and Delete event handlers, is so that if a user manually updates or deletes them, we re-create them per the spec that's defined in the NginxProxy resource. It requires a little bit more work than just setting the object here.

See `internal/controller/provisioner/store.go` and `internal/controller/provisioner/setter.go` to ensure that the object is handled properly.